### PR TITLE
proof for `make_zCDP_to_approxDP`

### LINKED ISSUE #524: `make_zCDP_to_approxDP` proof
Proof for #504.

### PR DESCRIPTION
**Pull request by Shoeboxam**
Closes #524.

Unblocks #505.

### CODE REVIEW

**Comment by Shoeboxam** [comment]
This PR needs to be updated to use the macros and appeal more closely to the trait definitions.

**Review by andrewvyrros — APPROVED** [review]
Approving with the understanding that this is still under the contrib flag. We'll need to make another pass over these for final approval, and resolution of the outdated marker.

**Pull request merged**
